src/EncumberableToken.sol

```diff
@@ -130,6 +130,7 @@ contract EncumberableToken is ERC20, IERC20Permit, IERC7246 {
         uint256 newEncumbrance = currentEncumbrance - amount;
         encumbrances[owner][taker] = newEncumbrance;
         encumberedBalanceOf[owner] -= amount;
+        emit EncumbranceSpend(owner, taker, amount);
```

Comment:
one quirk of this current implementation is that this event gets emitted even if the `amount` of the encumbrance is 0.

we should probably only be emitting this event if the encumbered amount is greater than 0

Reply:
That should only happen when someone does as `transferFrom` with 0 as the amount right? Probably not worth checking for this specific case.